modules/hitlnext/src/backend/repository.ts

```diff
@@ -62,6 +63,14 @@ export default class Repository {
     return object
   }
 
+  // This mutates object
+  private castJson(object: object, paths: string[]) {
```

Comment:
I'm not a fan of the naming here. I'd prefer `serialize` as this is only used for writing json to DB.

and does it have to mutate ?

Reply:
Agreed, renaming and removing mutation behavior.